lib/restforce.rb

```diff
@@ -49,7 +49,15 @@ module Tooling
   UnauthorizedError   = Class.new(Faraday::ClientError)
   APIVersionError     = Class.new(Error)
   BatchAPIError       = Class.new(Error)
-  CompositeAPIError   = Class.new(Error)
+
+  class CompositeAPIError < Error
```

Comment:
We should figure out the latest pattern in the Restforce library. I see some custom errors extend [Faraday Errors](https://github.com/lostisland/faraday/blob/main/lib/faraday/error.rb) which gives you an underlying response attribute. This line in [raise_error.rb](https://github.com/restforce/restforce/blob/10b22e6b10b7cca876d84b1ea2734d0478b4720b/lib/restforce/middleware/raise_error.rb#LL33C50-L33C50) also seems to suggest response_values may already be attached to errors

Reply:
good find. I can simply do `CompositeAPIError   = Class.new(Faraday::ClientError)` and attach response as a second param. As for the [raise_error.rb](https://github.com/restforce/restforce/blob/10b22e6b10b7cca876d84b1ea2734d0478b4720b/lib/restforce/middleware/raise_error.rb#LL33C50-L33C50) & Restforce middleware, Composite Response returns 200 even when there is a 400 response for subrequest so it skips this middleware and is directly handled on [composite_api.rb](https://github.com/restforce/restforce/pull/765/files#diff-652b9108b81ff7406ad380bfc726d290ecc4bd37613f0a83dfc18d0b7d3abd33R32).